vmdb/lib/miq_automation_engine/engine/miq_ae_workspace.rb

```diff
@@ -43,7 +44,7 @@ class MiqAeWorkspaceRuntime
     def initialize(options = {})
       options            = DEFAULTS.merge(options)
       @readonly          = options[:readonly]
-      @graph             = MiqAeDigraph.new
+      @nodes             = []
```

Comment:
No, this doesn't undo it.  The change I made [made `find_by_data` O(1)](https://github.com/ManageIQ/manageiq/commit/2ef1fed#diff-387f53928fe366a9010046c8543ba5b7R29).  `find_by_data` was used for doing the "1 to 1" lookup that I am talking about.  By turning `MiqAeObject` in to the node, we don't need to do that lookup anymore.  The old code [still kept track of every node](https://github.com/ManageIQ/manageiq/commit/2ef1fed#diff-387f53928fe366a9010046c8543ba5b7R10) but I'm not sure why.